### DISCUSSION TITLE
Análisis de Requerimientos

### DISCUSSION THREAD

**Discussion by carlos-gmx3** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#103] [Show and tell]
Como aporte al tema de Levantamiento de Requerimientos les vengo a compartir unos apuntes (se encuentran al final)
**Requerimientos Funcionales**
Como tal, son aquellos relacionados al proceso principal de un sitio web, si el objetivo es vender yoyos, para lograr este objetivo se necesita un catalogo de yoyos un carrito de compras un formulario que recolecte datos del cliente como la dirección de envio y una pasarela de pago en la parte de front
Para el back se necesita una base que tenga las existencias del inventario de yoyos y que registre los datos del cliente para que el vendedor pueda hacer el envío.

**Requerimientos no-funcionales**
Aquí pueden ir practicas adicionales, relacionadas al rendimiento de la pagina, como optimizarla para que cargue en poco tiempo, que la pagina pueda verse en dispositivos moviles y computadoras de forma optima e incluso aquellas relacionadas al contenido como una sección con tutoriales de trucos de yoyo e incluso un blog de novedades

_Tambien les dejo un ejercicio que realice en la escuela acerca del levantamiento de requerimientos y análisis de viabilidad de un proyecto_
**Entrevista**
Entrevistada: Wendy en representación de la clienta.

¿Qué esperas de una página que pueda administrar las citas con los clientes para tu trabajo?
Que me muestre cuando hay citas y a qué hora, y que si necesito escoger una fecha que me indique si está disponible o no y la duración de la misma y claro, el medio por el que pueda pagar.

¿Como vas a promocionar tu página?
Por medio de redes sociales y anuncios publicitarios y que me funcione también por recomendación

¿Quiénes podrán comprar en tu página?
Personas mayores a 18 años

¿La tienda estaría abierta al público en general o necesitan registrarse?
Abierta al público en general.

¿Qué datos necesitan ingresar para el informe de la reservación?
Nombre, correo, edad, sexo y teléfono

¿Cómo te gustaría agregar la disponibilidad de las citas?
En una cuadricula calendarizada donde pueda darle click al día y agregar la hora para la cual estará disponible.

¿Trabajaras por tu cuenta o con algún equipo?
Por mi cuenta

¿Qué funciones adicionales necesitas?
Ver estadísticas, de cuantas personas están ingresando, genero, edad, locación y poder incorporar otro tipo de servicios a la pagina

¿Necesitas incorporar un blog a la página?
Si me gustaría

¿Los métodos de pago como Pay Pal y mercado pago están bien?
Ujum, sí.

¿Tienes alguna idea de cómo te gustaría la página en diseño?
Que tenga un banner con los menús, incorporar información sobre la empresa, una slider con las imágenes de las entradas de blog, una sección de servicios adicionales.

¿Qué prioridad tiene que, en un futuro se integren artículos a la venta en la tienda?
Alta

¿Al finalizar una reservación, como quieres que se informe a los involucrados?
Mensaje al celular y correo electrónico, con la opción de revisar la agenda en la misma pagina.
 
**Requerimiento Funcionales**
Calendario dinámico: Tienda de citas y tablero de control manejaran un calendario, al dar click en los días mostrará la agenda del día. 

•	Para el cliente: Mostrará únicamente las citas disponibles para reservación

•	Para el vendedor: Mostrará la agenda del día que incluye, las citas que liberó para reservación, las ya reservadas y un botón para generar una cita disponible.

Generador de citas: Dentro de cualquier día del calendario dinámico, el vendedor podrá generar una nueva cita.

•	Crear Cita: Abrirá un recuadro en donde el vendedor podrá ingresar la hora del día en la cual podría atender a algún cliente, de igual forma podrá agregar la duración de la cita, siendo el valor por defecto de 1hr y un botón de aceptar, al hacer click en este, el sistema verificará que las citas generadas no interfieran en sus horarios unas con otras.

•	Eliminar cita: En caso de que el vendedor lo necesite, podrá eliminar cualquier cita que no haya sido reservada.

Carrito de Compras: Citas, productos o servicios serán agregados a un único carrito de compras que llevará la cuenta, esté estará disponible dentro del banner principal, por lo que un pequeño resumen podrá ser visualizado en cualquier parte del sitio web.

•	Agregar al carrito: De forma local, objetos como citas, productos o servicios serán almacenados en el carrito, sin embargo, ninguno de estos quedará reservado hasta que se haya realizado el pago.

•	Eliminar: Borrará del carrito y de la cuenta total el articulo seleccionado por el cliente. 

•	Finalizar Pedido: Llevará al cliente a un resumen de los artículos que eligió, el monto por articulo y el total, en caso de que un producto físico esté entre sus artículos, se le solicitará una dirección para el envío del artículo. 

Sistema de cobro: Cuando el cliente haya terminado de elegir los artículos que desea comprar será redirigido al sistema de pago de su elección.

•	Proceder al Pago: Ya sea Mercado Pago o PayPal, los sistemas de cobro serán manejados en el sitio a través de sus respectivas APIs donde se les proporcionara el monto que se debe pagar, quedando la página a la espera de la confirmación de que el pago ha sido realizado.

Entradas de Blog: El cliente podrá agregar entradas de Blog en cualquier momento, la imagen que el usuario agregue podrá incorporarse al slider de la página principal.

•	Agregar Entrada: Enviará al usuario a una página en donde podrá redactar un artículo para ser publicado en la página, campos como una imagen para el articulo y un título serán obligatorios para su publicación. Las entradas podrán ser compartidas en redes.

**No Funcionales**

Banner con enlaces y menús: Visible en cualquier página del sitio, será la herramienta principal de navegación, cuando se pase el mouse por encima, los vínculos que lo necesiten desplegaran un menú.

•	Reservaciones: Enviará directamente a la página del calendario para checar disponibilidad y solicitar una reservación.

•	Productos: Enviará a la página destinada a la venta de productos físicos.

•	Servicios: Enviará a la página donde estarán disponibles servicios que no dependan de una cita en tiempo real.

•	Carrito de compras: Desplegará un pequeño menú con los artículos que se han agregado al carrito con el total de la suma y un botón para finalizar el pedido.

Pantone para el sitio: El color principal para los colores que se usaran en el diseño es el morado: Matiz: 140, Saturación: 36, Luminosidad: 138.

Ecommerce: Incluir en el footer de la página los enlaces a redes sociales que el usuario indique, así como la posibilidad de compartir las entradas de blog en el muro de los clientes.

Slider en la página principal: Se mostrará de forma automática con intervalos de 5 segundos entre las imágenes. Incluirá:

•	Fotos de las últimas 3 entradas al blog.

•	Un anuncio de la sección de reservaciones.

•	Un anuncio con enlace a la red social principal del vendedor.

•	Hasta 3 anuncios para artículos a la venta.

Sitio Web Responsive: Dependiendo de la resolución de la cual se esté visualizando el sitio este se reestructurará para que pueda ser visualizados de una forma eficiente.

•	Versión Móvil: Cambiara el banner principal por uno más corto con un icono de hamburguesa para desplegar los enlaces. Se usará una sola columna para el contenido de la página.

•	Versión para Tablet: Organizará el contenido en una versión ajustada de la versión de computadora, se usará un banner mediano.

•	Versión para Computadoras de escritorio: Versión con todas las prestaciones, imágenes de tamaño completo y centrado de los elementos de la página, se usará un banner de tamaño completo.
 
**Estudio de Viabilidad**
El estudio de viabilidad aplicado al proyecto, es el siguiente.

Alcance
La plataforma será un complemento adicional a un canal de YouTube, en donde se va a promover su uso en caso de que los suscriptores quieran una atención personalizada.
Aunque dentro de la página también se requiere vender productos físicos, el foco se centrara en las citas a solicitud del cliente ya que, vía mensaje privado es lo que más solicita la gente que sigue el canal de YouTube.

Valoración de situación actual
Actualmente el canal de YouTube que va a promocionar estos servicios de forma adicional al contenido que se sube en dicha plataforma cuenta con más de cien mil subscritores y según reportes del cliente al día de hoy se atienden entre 2 y 4 citas al día de entre todos los mensajes que recibe solicitando una cita, Por lo que se espera un flujo moderado de visitas.

Por otra parte, en redes sociales como Facebook, Twitter e Instagram se pretende compartir las entradas al Blog, tanto en publicaciones como en la sección de historias de dichas redes sociales por lo que se necesita un footer dedicado a esta página del sitio donde invite a los visitantes a explorar otras áreas del sitio.

Definir los requisitos del sistema
Se Requiere de un servidor para alojar el sitio y el correo del mismo, los derechos de un dominio de internet tanto para el sitio como para los correos electrónicos de esta, al menos 10 Gb de almacenamiento para los archivos del sitio tanto el código y ejecutables que necesitara como el contenido multimedia que se desplegara en este.

Alternativas de Solución
1.	Adquirir una computadora y usarla como servidor.
2.	Adquirir un servidor
3.	Contratar un servicio de hosting

Valoración de las alternativas
Para el flujo inicial de visitas que se espera una computadora no es opción

Adquirir un servidor que de el ancho no es viable debido a la inversión que se requiere tanto para el equipo como el tiempo que se requiere para prepararlo, sin mencionar que se requiere un internet de banda ancha para que pueda atender múltiples solicitudes a la vez.

Contratar un servicio de hosting es económico y escalable, si el plan contratado no es suficiente se puede mejorar cambiando de plan.

Selección de la solución
El servicio de HostGator ofrece planes completos que no solo incluyen el servicio de hosting, sino también el derecho a un dominio personalizado con precios asequibles, en especial para proyectos que van iniciando por lo que parece la opción mas viable y utilizando la herramienta Google Analytics así como las que ofrece cPanel de HostGator, se puede monitorear el rendimiento del servicio contratado y mejorarlo si es necesario.

**Obtención y Análisis de Requerimientos**
Para esta fase de estandarización se realizó una entrevista con la clienta quien lleva una canal de YouTube, en esta entrevista ella expresó sus necesidades y prioridades para el sitió que requiere,

**Especificación de Requerimientos**
Para la especificación de requerimientos se dividió entre funcionales y no funcionales detallando de forma simple el comportamiento de las tareas del sistema, así como de los eventos que activaran dichos comportamientos. 
**Validación de Requerimientos**
Hasta este punto no se ha aplicado una validación más allá de verificar la redacción de los requerimientos, esto se podría hacer en dos pasos, una validación por parte del cliente donde pueda hacer correcciones o expresar sus dudas acerca de la compilación de requerimientos hasta el momento y dos, que se definan las restricciones referentes a el hosting del sitio que el cliente elija, así como la tecnología implementada para FrontEnd y BackEnd.
**Restricciones del Proyecto**
Tecnologías usadas
HTML, CSS, JavaScript con React para FrontEnd.
Java, MySQL y Rest para Backend.
Servicio de Hosting
25Gb de almacenamiento

Lo relacionado a la clasificación de requerimientos, es decir, funcionales y no-funcionales lo encuentran en la pagina 20
aunque todo el documento esta relacionado al tema de requerimientos en el área de desarrollo de software
_[Apuntes de Informática II (Administration de Requerimientos)](http://fcasua.contad.unam.mx/apuntes/interiores/docs/20181/informatica/2/LI_1216_05067_A_InformaticaII_Plan2016.pdf)_

**Fuentes de Consulta**

El estudio de viabilidad. (2016, 16 junio). [Archivo de vídeo]. Recuperado el 25 de marzo de 2021, de https://www.youtube.com/watch?v=FFtwcrpci3c

ESAN Graduate School of Business. (s. f.). ¿Cómo realizar un estudio de viabilidad de proyectos de manera correcta? Recuperado el 24 de marzo de 2021, de https://www.esan.edu.pe/apuntes-empresariales/2020/07/como-realizar-un-estudio-de-viabilidad-de-proyectos-de-manera-correcta-1/

Guerra, C. A. (s. f.). Obtención de Requerimientos. Técnicas y Estrategia. Recuperado el 25 de marzo de 2021, de https://sg.com.mx/revista/17/obtencion-requerimientos-tecnicas-y-estrategia

ITM Platform. (2019, 2 julio). Ventajas e inconvenientes de metodologías Ágil y Predictiva. Recuperado el 25 de marzo de 2021, de https://www.itmplatform.com/es/blog/ventajas-e-inconvenientes-de-metodologias-agil-y-predictiva/

Ok Diario. (2019, 28 enero). Cómo hacer un estudio de viabilidad. Recuperado el 24 de marzo de 2021, de https://okdiario.com/howto/como-hacer-estudio-viabilidad-3619683

Pérez, A. (2021, 15 marzo). Estudio de viabilidad de un proyecto: cómo y por qué llevarlo a cabo. Recuperado el 24 de marzo de 2021, de https://www.obsbusiness.school/blog/estudio-de-viabilidad-de-un-proyecto-como-y-por-que-llevarlo-cabo

**Comment by roge64** [comment]
Gracias por tu aportación Carlos, es de gran ayuda.

**Comment by Jimdrer** [comment]
Muy completo tu aporte compañero, muchísimas gracias por compartirlo, servirá de ayuda.

**Comment by LuceroLuciano** [comment]
Muchas gracias explorer, me ha ayudo bastante a entender la diferencia entre requerimientos funcionales y no funcionales.

**Comment by GloriaCoralCerecedo** [comment]
Muchas gracias, me esta ayudando

**Comment by jonaRL01** [comment]
¡muchas gracias!, he estado perdido estos dos días leyendo sobre todo y  solo me enredaba un poco mas.

**Reply by GloriaCoralCerecedo** [reply]
No te preocupes haz lo que puedas y lo que hayas entendido

**Comment by DavidSotoOrtiz** [comment]
¡Gracias Carlos!
